spec/acceptance/swap_file_class_spec.rb

```diff
@@ -2,49 +2,59 @@
 
 require 'spec_helper_acceptance'
 
-describe 'swap_file class', unless: UNSUPPORTED_PLATFORMS.include?(fact('osfamily')) do
-  context 'swap_file' do
-    context 'ensure => present' do
-      it 'works with no errors' do
-        pp = <<-EOS
-        class { 'swap_file':
-          files => {
-            'swapfile' => {
-              ensure => 'present',
-            },
-            'use fallocate' => {
-              swapfile => '/tmp/swapfile.fallocate',
-              cmd      => 'fallocate',
-            },
-            'remove swap file' => {
-              ensure   => 'absent',
-              swapfile => '/tmp/swapfile.old',
-            },
-          },
-        }
-        EOS
-
-        # Run it twice and test for idempotency
-        apply_manifest(pp, catch_failures: true)
-        apply_manifest(pp, catch_changes: true)
-      end
-
-      it 'contains the default swapfile' do
-        shell('/sbin/swapon -s | grep /mnt/swap.1', acceptable_exit_codes: [0])
-      end
+describe 'swap_file class' do
+  after(:all) do
+    pp = <<-EOS
+    class { 'swap_file':
+      files => {
+        'swapfile' => {
+          ensure => 'absent',
+        },
+        'use fallocate' => {
+          swapfile => '/tmp/swapfile.fallocate',
+          cmd      => 'fallocate',
+          ensure   => absent,
+        },
+        'remove swap file' => {
+          ensure   => 'absent',
+          swapfile => '/tmp/swapfile.old',
+        },
+      },
+    }
+    EOS
 
-      it 'contains the default fstab setting' do
-        shell('cat /etc/fstab | grep /mnt/swap.1', acceptable_exit_codes: [0])
-        shell('cat /etc/fstab | grep defaults', acceptable_exit_codes: [0])
-      end
+    apply_manifest(pp, catch_failures: true)
+  end
 
-      it 'contains the default swapfile' do
-        shell('/sbin/swapon -s | grep /tmp/swapfile.fallocate', acceptable_exit_codes: [0])
+  context 'with multiple files config' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-PUPPET
+          class { 'swap_file':
+            files => {
+              'swapfile' => {
+                ensure => 'present',
+              },
+              'use fallocate' => {
+                swapfile     => '/tmp/swapfile.fallocate',
+                cmd          => 'fallocate',
+                swapfilesize => '100MB'
+              },
+              'remove swap file' => {
+                ensure       => 'absent',
+                swapfile     => '/tmp/swapfile.old',
+                swapfilesize => '100MB'
+              },
+            },
+          }
+        PUPPET
       end
+    end
 
-      it 'contains the default fstab setting' do
-        shell('cat /etc/fstab | grep /tmp/swapfile.fallocate', acceptable_exit_codes: [0])
-      end
+    describe file('/etc/fstab'), shell('/sbin/swapon -s') do
```

Comment:
I don't think you can combine the `shell` part in here.

Actually, our code doesn't use this a lot but it's recommended to avoid `describe $resource()` and use:
```ruby
it 'sets up fstab' do
  expect(file('/etc/fstab')).to be_file
    .and(have_attributes(content: include('/mnt/swap.1')))
end
```

Reply:
and if I want to check the output of a shell script?